tickets/tickets_bot.go

```diff
@@ -210,8 +208,28 @@ func closeTicket(gs *dstate.GuildSet, currentTicket *Ticket, ticketCS *dstate.Ch
 		closingTicketsLock.Unlock()
 	}()
 
+	closingMsg := "Closing ticket."
+
+	// We only need to build up a more detailed closing msg.
+	// if we're creating logs.
+	if conf.TicketsUseTXTTranscripts || conf.DownloadAttachments {
+		var closingMsgBuilder strings.Builder
+		closingMsgBuilder.WriteString(closingMsg)
+
+		if conf.TicketsUseTXTTranscripts {
+			closingMsgBuilder.WriteString("\nCreating logs.")
+		}
+
+		if conf.DownloadAttachments {
+			closingMsgBuilder.WriteString("\nDownloading attachments.")
+		}
```

Comment:
The syntax of the generated message is a bit awkward:
```
Closing ticket.
Creating logs.
Downloading attachments.
This may take a while if the ticket is long.
```
I wonder if we can arrange for the message to be more similar to the original. Thoughts?

Reply:
I felt the *"and so on"* read weirdly, as those are the only two actions it's doing, and the comma version reads weirdly without it.

I could have tried to do a version conditionally inserting *"and"*, but this seemed like the least complex approach...

Reply:
Fair enough. For completeness, an alternative that is not too complicated is a switch enumerating all the cases, which is not too difficult here:
```go
transcripts, attachments := conf.TicketsUseTXTTranscripts, conf.DownloadAttachments
var closingMsg string
switch {
case transcripts && attachments:
	closingMsg = "Closing ticket, creating logs, and downloading attachments. [...]"
case transcripts:
	...
...
}
```
but it is arguable whether this is a real improvement.

Reply:
From the standpoint of minimising allocations, it's almost certainly an improvement. I was thinking of making a change of growing the allocation before building the string, but choosing a single option is almost certainly better,

Reply:
It is definitely more performant, but this code is not on the hot path and we do not make much effort to minimize allocations in general so that concern may be misplaced. The main questions, I think, are quality of the output message, code readability, and extensibility, and it's a bit less clear whether the above suggestion is a meaningful improvement on all three fronts to be worthwhile--I don't have a strong opinion.

Reply:
Maybe more notably in that case: The switch statement can make for formatting more intuitive...

I previously noted how it would be awkward to insert "and"s into the resulting message, but the switch statement would make directly using those responses easier.

Reply:
Alternatively: The initial line could be added as three statements;
`Closing ticket. Creating logs. Downloading attachments.`

Having looked at my solution, using a `switch` statement, and string builder, it doesn't look better to me...

Reply:
What about a slice of actions joined by commas and switched to title case
at the end?

Reply:
I don't particularly see how that helps. It would create a single source for changing any separators, yes, but would largely result in checking the same conditions anyway...

Reply:
Just joins it all into one sentence instead of three separate ones.
maybe I’m not understanding what the current issue is.

Reply:
The initial quibble, was about the fact I changed the response to be separated onto different lines, and the message reading somewhat strangely as a result. I kind of agree with that in hindsight, and I think the three statements could be on the same line.

So basically, just swapping `\n` for ` `